### PR TITLE
Use Shamir as KeK when migrating from auto-seal to shamir

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: command/seal_migration_test.go [hunk 1]
```diff
@@ -5,21 +5,93 @@ package command
...
+	core := cluster.Cores[0].Core
+
+	shamirSeal := vault.NewDefaultSeal(&seal.Access{
+		Wrapper: aeadwrapper.NewWrapper(&wrapping.WrapperOptions{
+			Logger: logger.Named("shamir"),
+		}),
+	})
+	shamirSeal.SetCore(core)
+
+	if err := adjustCoreForSealMigration(logger, core, shamirSeal, autoSeal); err != nil {
```
[the remark this answers is not in the document]
Comment (vishalnayak) [reply command/seal_migration_test.go]: The file already has those build flags.

File: vault/core.go [hunk 2]
```diff
@@ -1273,6 +1273,15 @@ func (c *Core) unsealPart(ctx context.Context, seal Seal, key []byte, useRecover
 				return nil, errors.New("did not get expected recovery information to set new seal during migration")
 			}
 
+			if err := c.seal.SetBarrierConfig(ctx, &SealConfig{
+				Type:            wrapping.Shamir,
+				SecretShares:    1,
+				SecretThreshold: 1,
+				StoredShares:    1,
```
Comment (briankassouf) [inline vault/core.go]: I think you may want something along the lines of:

```
				Type:            wrapping.Shamir,
				SecretShares:    recoveryConfig.SecretShares,
				SecretThreshold: recoveryConfig.SecretThreshold,
				StoredShares:    1,
```

Comment (vishalnayak) [reply vault/core.go]: Done. Thanks to @ncabatoff!

File: command/seal_migration_test.go [hunk 3]
```diff
@@ -5,21 +5,97 @@ package command
...
+		SkipInit:    true,
+		NumCores:    1,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	client := cluster.Cores[0].Client
+	resp, err := client.Sys().Init(&api.InitRequest{
+		RecoveryShares:    1,
+		RecoveryThreshold: 1,
```
Comment (briankassouf) [inline command/seal_migration_test.go]: Maybe do a shares:5, threshold:3 to verify the config is being updated correctly?

File: command/seal_migration_test.go [hunk 4]
```diff
@@ -5,21 +5,97 @@ package command
...
+		if err != nil {
+			t.Fatal(err)
+		}
+		if resp == nil {
+			t.Fatal("expected response")
+		}
+	}
+	if statusResp.Sealed {
+		t.Fatalf("expected unsealed state; got %#v", *resp)
+	}
```
Comment (briankassouf) [inline command/seal_migration_test.go]: Maybe do a seal/unseal here to make sure everything has been updated appropriately?

File: command/seal_migration_test.go (hunk 3 above)
Comment (vishalnayak) [reply command/seal_migration_test.go]: done

File: command/seal_migration_test.go (hunk 4 above)
Comment (vishalnayak) [reply command/seal_migration_test.go]: done.

**Pull request merged**